File: backend/app/api/v1/endpoints/roles.py
from typing import List
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.ext.asyncio import AsyncSession
from app.core.database import get_db
from app.core.deps import get_current_active_user
from app.models.user import User

router = APIRouter()

@router.get("/")
async def get_roles(
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    return []

@router.get("/{role_id}")
async def get_role(
    role_id: str,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    return {}

@router.post("/")
async def create_role(
    role_data: dict,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    return role_data

@router.put("/{role_id}")
async def update_role(
    role_id: str,
    role_data: dict,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    return role_data

@router.delete("/{role_id}")
async def delete_role(
    role_id: str,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
):
    return {"message": "Role deleted"}

File: frontend/src/api/roles.ts
import request from '@/utils/request'
import type { Role } from '@/types'

export const rolesApi = {
  // Get all roles
  getRoles() {
    return request.get<Role[]>('/v1/roles/')
  },

  // Get role by ID
  getRole(roleId: string) {
    return request.get<Role>(`/v1/roles/${roleId}`)
  },

  // Create role
  createRole(data: Partial<Role>) {
    return request.post<Role>('/v1/roles/', data)
  },

  // Update role
  updateRole(roleId: string, data: Partial<Role>) {
    return request.put<Role>(`/v1/roles/${roleId}`, data)
  },

  // Delete role
  deleteRole(roleId: string) {
    return request.delete<{ message: string }>(`/v1/roles/${roleId}`)
  },
}
